### ISSUE TITLE
Yo Dawg, I Heard you Like Templates

### ISSUE THREAD

**Issue by cholick** [cf-platform-eng/tile-generator#199]
Currently, when deploying a package of type app, the manifest as defined in `tile.yml` _does not_ go through Ops Managers templating substitution. This prevents form fields from controlling manifest values.

Use case: the deployed app's number of instances needs to be scaled out depending on the foundation's size (a nozzle, for example)

**Comment by cholick** [comment]
Workaround:
```
---
name: tile-gen-environment-check
icon_file: icon.png
label: Test tile
description: Cholick test tile

packages:
- name: tile-test-app-cholick
  type: app
  manifest:
    instances: MY_INSTANCE_COUNT
    path: app.zip
    buildpack: python_buildpack
    command: python app.py
    memory: 256M
  pre_deploy: |
    sed -i "s/MY_INSTANCE_COUNT/$SCALE_OUT/g" $PACKAGE_PATH/tile_test_app_cholick/manifest.yml


forms:
- name: test-config
  label: Test Config
  properties:
  - name: sample_property
    type: string
    label: Sample property
    description: Sample property to validate incoming app environment
  - name: scale_out
    type: integer
    label: Instances to run
    description: Always run 2
    default: 2
```

**Comment by mboldt** [comment]
Maybe the right answer is to have have the app manifest file use bosh erb templates, so we could do something like:
```
instances: <%= p('scale_out') %>
```

**Comment by mboldt** [comment]
After further discussion, we will put the app metadata into a property in the job manifest in the deploy-all errand. The deploy-all errand can read that property and write it to a file for the cf push. That will allow Ops Manager to do `(( ... ))` substitution on the app metadata. (Clear as mud?)

**Comment by mboldt** [comment]
Released in 9.0.4. Sad to see this issue close. I will forever refer to the trick of shoving bits through Ops Manager properties so they can be templated as "yo dawg".
